### ISSUE TITLE
Enable variable explorer when autocalling pdb

### ISSUE THREAD

**Issue by cmeeren** [spyder-ide/spyder#2789]
## Description

**What steps will reproduce the problem?**
1. Turn on pdb autocalling by typing `pdb` in the iPython console
2. Run a script which sets a variable and then fails
3. Observe that the variable explorer is empty

Example script:

```
    foo = 2+2
    bar = 1/0
```

**What is the expected output? What do you see instead?**

I expect `foo` to be visible in the variable explorer. Instead, nothing is visible.
## Version and main components
- Spyder Version: 2.3.7 
- Python Version: 2.7.10
- Qt Versions:  4.8.6, PyQt4 (API v2) 4.10.4 on Windows
## Optional dependencies

```
pyflakes >=0.5.0   :  0.8.1 (OK)
pep8 >=0.6         :  1.6.2 (OK)
pygments >=1.6     :  2.0.2 (OK)
IPython >=1.0      :  3.0.0 (OK)
zmq >=2.1.11       :  14.5.0 (OK)
pandas >=0.13.1    :  0.16.1 (OK)
sphinx >=0.6.6     :  1.2.3 (OK)
rope >=0.9.2       :  0.9.4 (OK)
jedi >=0.8.1;<0.9.0:  0.8.1 (OK)
matplotlib >=1.0   :  1.4.2 (OK)
sympy >=0.7.3      :  0.7.6 (OK)
pylint >=0.25      :  1.4.1 (OK)

```

**Comment by ccordoba12** [comment]
We use our own patched version of `pdb` to allow connections to the Variable Explorer. It seems that version is not the one called by IPython when typing `pdb` on their consoles.

We'll see what we can do about it :-)

**Comment by ccordoba12** [comment]
This was fixed in Spyder 3.2
